### PR TITLE
Pathfinding update

### PR DESCRIPTION
**Pull request by Westlanderz**
This PR will update the pathfinding so that players will move around other players. Please test  this before merging to make sure I did not forget any edge-cases.

### CODE REVIEW

File: Assets/Scripts/Core/GridCombatSystem.cs [hunk 1]
```diff
@@ -149,17 +149,22 @@ private enum State {
...
 				for(int y = unitY - maxMoveDistance; y <= unitY + maxMoveDistance; y++) {
 					if(x < 0 || x >= grid.GetWidth() || y < 0 || y >= grid.GetHeight()) {
 						continue;
 					}
-					if(GameController.Instance.grid.GetGridObject(x, y).GetUnitGridCombat() != null) {
-						continue;
-					}
 
-					if(gridPathfinding.IsWalkable(x, y)) {
+					if(gridPathfinding.IsWalkable(x, y) && x != unitX || y != unitY) {
```
Comment (drfdaniel) [inline Assets/Scripts/Core/GridCombatSystem.cs:167]: What was unitX & unitY again? I assume current position?

**Review by drfdaniel — DISMISSED** [review]

File: Assets/Scripts/Core/GridCombatSystem.cs (hunk 1 above)
Comment (Westlanderz) [reply Assets/Scripts/Core/GridCombatSystem.cs:167]: Those are the x and y coords of the selected unit.

**Review by Dennisbonke — APPROVED** [review]
LGTM
